workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs

```diff
@@ -268,7 +268,11 @@ private void AddCoreSystems()
 
         public void Dispose()
         {
-            World?.Dispose();
+            if (World != null && World.IsCreated)
+            {
+                World?.Dispose();
```

Comment:
No need for the null conditional operator since we explicitly do the null check 😄